doc/release/README.chplenv.rst

```diff
@@ -0,0 +1,555 @@
+        ======= ============================================
+
+   If unset, CHPL_COMM defaults to "none" in most cases.  On a Cray system it
+   defaults to "gasnet" except when using a pre-built module on a Cray XC or XE
+   (TM) system, where it defaults to "ugni".  See README.multilocale for more
+   information on executing Chapel programs using multiple locales.  See also
+   platforms/README.cray for more information about Cray-specific runtime
+   layers.
+
+
+CHPL_MEM
+~~~~~~~~
+   Optionally, the CHPL_MEM environment variable can be used to select
+   a memory management layer.  Current options are:
+
+        ========= =======================================================
+        Value     Description
+        ========= =======================================================
+        cstdlib   use the standard C malloc/free commands
+        dlmalloc  use Doug Lea's memory allocation package
+        tcmalloc  use the tcmalloc package from Google Performance Tools
+        ========= =======================================================
+
+   If unset, CHPL_MEM defaults to "cstdlib" in most cases.  If comm is "gasnet"
+   and you are using the fast or large segments it defaults to "dlmalloc".  If
+   comm is "ugni" it defaults to "tcmalloc".  See README.multilocale for more
+   information on GASNet segments.
+
+
+CHPL_LAUNCHER
+~~~~~~~~~~~~~
+   Optionally, the CHPL_LAUNCHER environment variable can be used to
+   select a launcher to get your program up and running.  See README.launcher
+   for more information on this variable's default and possible settings.
+
+
+CHPL_ATOMICS
+~~~~~~~~~~~~
+   Optionally, the CHPL_ATOMICS environment variable can be used to
+   select an implementation for atomic operations in the runtime.  Current
+   options are:
+
+        ===========  =====================================================
+        Value        Description
+        ===========  =====================================================
+        intrinsics   implement atomics using target compiler intrinsics
+                     (which typically map down to hardware capabilities)
+        locks        implement atomics by using Chapel sync variables to
+                     protect normal operations
+        ===========  =====================================================
+
+   If unset, CHPL_ATOMICS defaults to "intrinsics" for most configurations.  On
+   some 32 bit platforms, or if the target compiler is "pgi" or
+   "cray-prgenv-pgi" it defaults to "locks".
+
+   .. note::
+     gcc 4.8.1 added support for 64 bit atomics on 32 bit platforms.  We
+     default to "intrinsics" for 32 bit platforms when using the target
+     compiler "gnu" with a recent enough version of gcc.  For older versions or
+     other target compilers we default to "locks"
+
+   See the Chapel Language Specification for more information about atomic
+   operations in Chapel or doc/technotes/README.atomics for more information
```

Comment:
This probably should point to the module documentation now (though I'm sure you would've caught that on your own :) )